spec/scenarios/charge_models/prorated_graduated_spec.rb

```diff
@@ -0,0 +1,399 @@
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '20' },
+              },
+            )
+
+            fetch_current_usage(customer:)
+            expect(json[:customer_usage][:amount_cents].round(2)).to eq(38935)
+            expect(json[:customer_usage][:total_amount_cents].round(2)).to eq(38935)
+            expect(json[:customer_usage][:charges_usage][0][:units]).to eq('95.0')
+          end
+
+          subscription = customer.subscriptions.first
+
+          travel_to(DateTime.new(2023, 10, 18)) do
+            create(
+              :graduated_charge,
+              billable_metric:,
+              prorated: true,
+              plan: plan_new,
+              properties: {
+                graduated_ranges: [
+                  {
+                    from_value: 0,
+                    to_value: 5,
+                    per_unit_amount: '10',
+                    flat_amount: '100',
+                  },
+                  {
+                    from_value: 6,
+                    to_value: 15,
+                    per_unit_amount: '5',
+                    flat_amount: '50',
+                  },
+                  {
+                    from_value: 16,
+                    to_value: nil,
+                    per_unit_amount: '2',
+                    flat_amount: '0',
+                  },
+                ],
+              },
+            )
+            expect {
+              create_subscription(
+                {
+                  external_customer_id: customer.external_id,
+                  external_id: customer.external_id,
+                  plan_code: plan_new.code,
+                },
+              )
+            }.to change { subscription.reload.status }.from('active').to('terminated')
+              .and change { subscription.invoices.count }.from(1).to(2)
+
+            invoice = subscription.invoices.order(created_at: :desc).first
+            expect(invoice.fees.charge_kind.count).to eq(1)
+            expect(invoice.total_amount_cents).to eq(30484) # 30226 + 2.58 (prorated event in termination period)
+          end
+
+          travel_to(DateTime.new(2023, 11, 1)) do
+            Subscriptions::BillingService.new.call
+
+            perform_all_enqueued_jobs
+
+            subscription = customer.subscriptions.order(created_at: :desc).first
+            invoice = subscription.invoices.order(created_at: :desc).first
+
+            aggregate_failures do
+              # (95 units * 14/31) -> 30581 - charge fee
+              # 100 * 14/31 -> 45 -> subscription fee
+              expect(invoice.total_amount_cents).to eq(30581 + 45)
+              expect(subscription.reload.invoices.count).to eq(1)
+            end
+          end
+
+          travel_to(DateTime.new(2023, 11, 5)) do
+            fetch_current_usage(customer:)
+            expect(json[:customer_usage][:amount_cents].round(2)).to eq(41000)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)